api.py

```diff
@@ -0,0 +1,71 @@
+from flask import Flask, request, jsonify, abort
+import pandas as pd
+import requests
+import json
+import os
+
+app = Flask(__name__)
+
+API_KEY = os.environ['API_SECRET']
+WEATHER_API_KEY = os.environ['WEATHER_API_SECRET']
+
+def require_token(func, *args, **kwargs):
+    def wrapper():
+        if 'Authorization' in request.headers and request.headers['Authorization'] == f'Bearer {API_KEY}':
+            return func(*args, **kwargs)
+
+        else:
+            abort(401, description="Error message you want the reader to receive.")
+        
+    return wrapper
+
+@app.route('/estimate', methods=['POST'])
+@require_token
+def estimate():
+    request_data = request.json
```

Comment:
1. parsing/data validation here 
2. Calling the weather api
3. Calculating results

```
# throws errors and sends validation errors back in json 
# can be in module validation
params = validation.validate_estimate_params(request_data) 

# In module services, params can be kwargs
weather_df = services.load_weather_data(params)

# Calculates energy production and profit from selling all energy
production = simulation.calculate_production(weather_df, params)
profit = simulation.calculate_profit(production)

# Serialize objects with serializer
# Ensures that the right data is returned
# Does something like below
# `json.dumps( {timestamp, production, profit})`
result = serializer.serialize_estimation_results(prodcution, profit)
return result
```